x-pack/plugins/elastic_assistant/server/routes/knowledge_base/post_knowledge_base.ts

```diff
@@ -49,46 +45,34 @@ export const postKnowledgeBaseRoute = (
         validate: {
           request: {
             params: buildRouteValidationWithZod(CreateKnowledgeBaseRequestParams),
+            query: buildRouteValidationWithZod(CreateKnowledgeBaseRequestQuery),
           },
         },
       },
-      async (
-        context,
-        request: KibanaRequest<CreateKnowledgeBaseRequestParams>,
-        response
-      ): Promise<IKibanaResponse<CreateKnowledgeBaseResponse>> => {
+      async (context, request, response): Promise<IKibanaResponse<CreateKnowledgeBaseResponse>> => {
         const resp = buildResponse(response);
         const ctx = await context.resolve(['core', 'elasticAssistant', 'licensing']);
         const assistantContext = ctx.elasticAssistant;
-        const logger = ctx.elasticAssistant.logger;
-        const telemetry = assistantContext.telemetry;
-        const elserId = await getElser();
         const core = ctx.core;
-        const esClient = core.elasticsearch.client.asInternalUser;
         const soClient = core.savedObjects.getClient();
+        const kbResource = request.params.resource;
+        const modelIdOverride = request.query.modelId;
 
         // FF Check for V2 KB
         const v2KnowledgeBaseEnabled = isV2KnowledgeBaseEnabled({ context: ctx, request });
 
         try {
           const knowledgeBaseDataClient =
-            await assistantContext.getAIAssistantKnowledgeBaseDataClient(v2KnowledgeBaseEnabled);
+            await assistantContext.getAIAssistantKnowledgeBaseDataClient({
+              modelIdOverride,
+              v2KnowledgeBaseEnabled,
+            });
           if (!knowledgeBaseDataClient) {
             return response.custom({ body: { success: false }, statusCode: 500 });
           }
 
-          // Continue to use esStore for loading esql docs until `semantic_text` is available and we can test the new chunking strategy
-          const esStore = new ElasticsearchStore(
-            esClient,
-            knowledgeBaseDataClient.indexTemplateAndPattern.alias,
-            logger,
-            telemetry,
-            elserId,
-            getKbResource(request),
-            knowledgeBaseDataClient
-          );
-
-          await knowledgeBaseDataClient.setupKnowledgeBase({ esStore, soClient });
+          const installEsqlDocs = kbResource === ESQL_RESOURCE;
+          await knowledgeBaseDataClient.setupKnowledgeBase({ soClient, installEsqlDocs });
```

Comment:
Trying to phase out `esStore` as much as possible so we can finally remove it. The `esql_loader` has been updated to load using the `kbDataClient` directly now instead of passing through `esStore` (it has been using the `kbDataClient` indirectly since `8.15`).